src/consumer.rs

```diff
@@ -1369,35 +1451,170 @@ impl<Exe: Executor> ConsumerEngine<Exe> {
                 self.options.clone(),
             )
             .await
-            .map_err(Error::Connection)?;
+        {
+            Ok(_success) => {
+                if *current_retries > 0 {
+                    let dur = (Instant::now() - start).as_secs();
+                    log::info!(
+                        "subscribing({}) success after {} retries over {} seconds",
+                        topic,
+                        *current_retries + 1,
+                        dur
+                    );
+                }
+            }
+            Err(ConnectionError::PulsarError(Some(err), text)) => {
+                return match err {
+                    proto::ServerError::ServiceNotReady | proto::ServerError::ConsumerBusy => {
+                        match operation_retry_options.max_retries {
+                            Some(max_retries) if *current_retries < max_retries => {
+                                self.retry_with_delay(
+                                    current_retries,
+                                    err.as_str_name(),
+                                    topic,
+                                    &text,
+                                    broker_address,
+                                )
+                                .await
+                                .map_err(Err)?;
+
+                                Err(Ok(()))
+                            }
+                            _ => {
+                                error!("subscribe topic({}) reached max retries", topic);
+
+                                Err(Err(ConnectionError::PulsarError(Some(err), text).into()))
+                            }
+                        }
+                    }
+                    _ => Err(Err(Error::Connection(ConnectionError::PulsarError(
+                        Some(err),
+                        text,
+                    )))),
+                }
+            }
+            Err(ConnectionError::Io(e)) => {
```

Comment:
Could be write to reduce a level of indentation like this:
```suggestion
            Err(ConnectionError::Io(e)) if matches!(e.kind(),ErrorKind::BrokenPipe | .. | ErrorKind::UnexpectedEof) => ..
```

Reply:
Sorry but I don't understand